### ISSUE TITLE
Feature request: expand humanization to bar/measure/pattern duration

### ISSUE THREAD

**Issue by github-wlk** [hydrogen-music/hydrogen#1168]
Hydrogen version *          :Future
Operating system + version  : all
Audio driver + version      : n/a

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Per https://github.com/hydrogen-music/hydrogen/discussions/1162 I'm wondering if there should be humanization (timing, specifically) added to the length of the bar/pattern/measure (i.e. wherever it is currently not)

True humanization would entail slight stretching or compressing of the time.  Let's say 4/4 and quarter=60...  Every 4.0000 seconds, you're not going to nail the "one" :)   Even if you're Colaiuta :)  Otherwise, it's almost more rubato, and you still nail that "one" :)   And if my assumptions about this are wrong (in the way H2 currently behaves), let me know!

So, just throwing the idea out there whether this should even be considered or not, or if there is any interest in it.  I wouldn't think it would necessarily need to be the default for timing humanization, specifically for legacy purposes.  Perhaps a checkmark in the preferences (default: off) for whether you want pattern-level timing variability.

Thanks!

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

*  H2 version : see 'Info - About', or in case you build Hydrogen from source use 'git rev-parse HEAD'
** if you are reporting a bug, please describe how the bug can be reproduced

**Comment by oddtime** [comment]
I think such humanization would be achieved with the pull req #627 by @theGreatWhiteShark

Oherwise I would tend to achieve what you are requesting with a variation in Bpm probably

**Comment by oddtime** [comment]
> True humanization would entail slight stretching or compressing of the time. Let's say 4/4 and quarter=60... Every 4.0000 seconds, you're not going to nail the "one" :) Even if you're Colaiuta :) Otherwise, it's almost more rubato, and you still nail that "one" :) And if my assumptions about this are wrong (in the way H2 currently behaves), let me know!

You could think the current implementation more like a human drummer playing with the click in the headphones

**Comment by theGreatWhiteShark** [comment]
> I think such humanization would be achieved with the pull req #627 by @theGreatWhiteShark

Nope. This is about having a more "human" spectrum of the noise added during the humanization. But having random walk in velocity, timing, and overall speed would indeed be the way to go. Randomization in timing is planned to be implement right after the velocity one. However, a random walk in the overall BPM might be not feasible in combination with the JACK server.
